packages/test/test-service-load/src/runner.ts

```diff
@@ -224,8 +224,15 @@ async function runnerProcess(
 			container.connect();
 			const test = await requestFluidObject<ILoadTest>(container, "/");
 
-			// Retain old behavior of runtime being disposed on container close
-			container.once("closed", () => container?.dispose());
+			const closeDisposeProm = new Deferred<boolean>();
```

Comment:
@vladsud This is along the lines of what I was describing. However, this won't work for signals in the test. Even though we hit the op counts, the signal count is not guaranteed to be met before we inject a nack or closure (which will cause the test to endlessly loop).

Likely the best solution is use what I already committed, in that whenever the host sees the container close it should dispose all resources and start over.

@anthony-murphy The idea is that the runtime should no longer be concerned whether the container closed or not (but we could create a task to have the runtime emit some new "closed" event so a consumer could observe as needed).
For example, say we have a scenario where container closes but application still allows user to scroll around a page and load blobs. Previously, this would not work as storage service (along with runtime) would be disposed.

For most cases (including this one), the host is responsible for reacting to the container closure. For this case, the host code knows that when the container closes we cannot proceed anymore with the current runtime.

The model that container can "dispose" itself does not feel right as someone may have a reference to the container or runtime and still want to do some work on it.